kernel/core/utils/tools.cpp

```diff
@@ -1980,85 +1981,59 @@ bool addLegends(string& sExpr)
 
     std::string sTemp;
 
-    for (size_t i = 0; i < args.size(); i++)
+    for (std::string arg : args)
     {
-        /* --> Nun koennte es sein, dass bereits eine Legende angegeben worden ist. Dabei gibt es drei
-         *     Moeglichkeiten: entweder durch umschliessende Anfuehrungszeichen, durch eine vorangestellte
-         *     Raute '#' oder auch beides. Wir muessen hier diese drei Faelle einzeln behandeln <--
-         * --> Ebenfalls ist es natuerlich moeglich, dass gar keine Legende angegeben worden ist. Das behandeln
-         *     wir im ELSE-Fall <--
-         */
-        if (args[i].find('"') != std::string::npos)
+        if (sTemp.length())
+            sTemp += ", ";
+
+        // Find the end of the current expression first
+        for (size_t i = 0; i < arg.length(); i++)
         {
-            /* --> Hier ist auf jeden Fall '"' vorhanden. Es ist aber nicht gesagt, dass '#' nicht auch
-             *     zu finden ist <--
-             * --> Speichern wir zunaechst die Position des '"' in nPos <--
-             */
-            size_t nPos = args[i].find('"') + 1;
-
-            // --> Pruefe nun, ob in diesem Stringbereich ein zweites '"' zu finden ist <--
-            if (args[i].find('"', nPos) != std::string::npos)
+            if (arg[i] == '{' || arg[i] == '(')
+                i += getMatchingParenthesis(StringView(arg, i));
+
+            if (arg[i] == '"')
             {
-                // --> Ja? Gibt's denn dann eine Raute? <--
-                if (args[i].find('#', nPos) != std::string::npos)
+                if (arg.find('"', i+1) != std::string::npos)
                 {
-                    // --> Ja? Dann muessen wir (zur Vereinfachung an anderer Stelle) noch zwei Anfuehrungszeichen ergaenzen <--
-                    sTemp += args[i] + "+\"\"";
+                    if (arg.find('#', i+1) != std::string::npos)
+                        sTemp += arg + "+\"\"";
+                    else
+                        sTemp += arg;
+
+                    break;
                 }
                 else
-                    sTemp += args[i];
+                    return false;   // Nein? Dann ist irgendwas ganz Falsch: FALSE zurueckgeben!

```

Comment:
I do not speak this language :P Please do se english

Reply:
Uff. Very old comment 😁